activesupport/lib/active_support/cache/strategy/local_cache.rb

```diff
@@ -75,7 +75,10 @@ def delete_entry(key, options)
           end
 
           def fetch_entry(key, options = nil) # :nodoc:
-            @data.fetch(key) { @data[key] = yield }
+            entry = @data.fetch(key) { @data[key] = yield }
+            dup_entry = entry.dup
+            dup_entry&.dup_value!
```

Comment:
How could `dup_entry` ever be nil?

Reply:
in case you try to access a key that doesn't exist

Reply:
Nevermind, I'm stuck in the past, turns out `nil.dup` actually exist on modern rubies TIL.

Reply:
I think this would have to be a deep copy. `.dup` alone will still let you modify nested objects if present

![image](https://user-images.githubusercontent.com/1896520/61063253-a5fdaf00-a3cd-11e9-8b66-2d282ecd75ca.png)

Reply:
No because `dup_value!` does a Marshal.load https://github.com/rails/rails/blob/457c815c93010b1884f675753daf4fc6950fb6af/activesupport/lib/active_support/cache.rb#L793